FHICORC.Backend/FHICORC.Application.Services/RevocationService.cs

```diff
@@ -53,6 +60,140 @@ public RevocationService(ILogger<RevocationService> logger, CoronapassContext co
+            {
+                expiresMatch = ExpiryDate == batchItem.Expires.Date;
+                batchItemCount = batchItem.Count;
+            }
+
+            foreach (var hash in newHashList.Select((value, index) => new { value, index }))
+            {
+                if (hash.index < (MAX_SIZE_BATCH - batchItemCount) && expiresMatch)
+                {
+                    CreateHash(batchItem.BatchId, hash.value);
+                }
+                else if (counter < MAX_SIZE_BATCH)
+                {
+                    if (firstIteration)
+                    {
+                        CreateBatch(newGuid.ToString());
+                        firstIteration = false;
+                        counter = 0;
+                    }
+                    CreateHash(newGuid.ToString(), hash.value);
+                    counter++;
+                }
+                else
+                {   
+                    newGuid = Guid.NewGuid();
+                    counter = 1;
+                    
+                    CreateBatch(newGuid.ToString());
+                    CreateHash(newGuid.ToString(), hash.value);
+                }
+            }
+        }
+
+        private void CreateHash(string batchId, string hash)
+        {
+            var hashDto = new RevocationHash
+            {
+                BatchId = batchId,
+                Hash = hash,
+            };
+            _coronapassContext.ChangeTracker.Clear();
+            _coronapassContext.RevocationHash.Add(hashDto);
+            _coronapassContext.SaveChanges();
+        }
+        private void CreateBatch(string batchId)
+        {
+            var newBatch = new RevocationBatch
+            {
+                BatchId = batchId,
+                Expires = DateTime.Now.AddMonths(3),
+                Country = COUNTRY_CODE,
+                HashType = HASH_TYPE,
+                Deleted = false,
+                Upload = true,
+            };
+            _coronapassContext.ChangeTracker.Clear();
+            _coronapassContext.RevocationBatch.Add(newBatch);
+            try
+            {
+
+                _coronapassContext.SaveChanges();
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError($"Duplicate Key: {batchId}, message: {ex.Message}");
+            }
+        }
+        private List<string> ReturnUniqueHashes(List<string> hashList, HashesDto revokedHashList)
+        {
+            return hashList
+                .Where(h => revokedHashList.Hashes
+                .All(rh => rh.HashInfo != h))
+                .ToList();
+        }
+        private HashesDto FetchHashes()
+        {
+            var hashList = _coronapassContext.RevocationHash
+                .Select(x => new Hash()
+                {
+                    Id = x.Id,
+                    BatchId = x.BatchId,
+                    HashInfo = x.Hash,
+                }
+                ).ToList();
+
+            return new HashesDto()
```

Comment:
Class not needed just return the list in an expression body:

private HashesDto FetchHashes() => _coronapassContext.RevocationHash.Select(x => new Hash(x));